includes/base/controls-stack.php

```diff
@@ -818,6 +818,13 @@ final public function add_responsive_control( $id, array $args, $options = [] )
 		$is_frontend_available = ! empty( $args['frontend_available'] );
 		$has_prefix_class = ! empty( $args['prefix_class'] );
 
+		if ( ! empty( $options['overwrite'] ) ) {
+			// If this is an updated control, check if it needs to be duplicated or not.
+			$is_duplicated_overwrite = $this->is_duplicated_overwrite( $id );
```

Comment:
Get the control to check with instead of duplicating the conditions

Reply:
Done.